libevm/EVMC.cpp

```diff
@@ -23,33 +23,38 @@ EVM::EVM(evm_instance* _instance) noexcept : m_instance(_instance)
 
 owning_bytes_ref EVMC::exec(u256& io_gas, ExtVMFace& _ext, OnOpFunc const& _onOp)
 {
-	bool rejected = false;
-	// TODO: Rejecting transactions with gas limit > 2^63 can be used by attacker to take JIT out of scope
-	rejected |= io_gas > std::numeric_limits<int64_t>::max(); // Do not accept requests with gas > 2^63 (int64 max)
-	rejected |= _ext.envInfo().number() > std::numeric_limits<int64_t>::max();
-	rejected |= _ext.envInfo().timestamp() > std::numeric_limits<int64_t>::max();
-	rejected |= _ext.envInfo().gasLimit() > std::numeric_limits<int64_t>::max();
-	if (rejected)
-	{
-		cwarn << "Execution rejected by EVM JIT (gas limit: " << io_gas << "), executing with interpreter";
-		return VMFactory::create(VMKind::Interpreter)->exec(io_gas, _ext, _onOp);
-	}
+    bool rejected = false;
+    // Do not accept requests with gas > 2^63 (int64 max)
+    rejected |= io_gas > std::numeric_limits<int64_t>::max();
```

Comment:
I think this should also test for `_ext.depth`. I know it is unlikely due to other restrictions, but `execute` does a typecast `static_cast<int32_t>(_ext.depth)` and it feels better to do this correctly.